tests/integration/Unticketed_Cost_Formatting_Test.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace Tribe\Events;
+
+class Unticketed_Cost_Formatting_Test extends \Codeception\TestCase\WPTestCase {
+
+	/**
+	 * Container for our test events. Each event will be stored as an array of
+	 * [ cost, post_id ] - in other words, this is an array of arrays.
+	 *
+	 * @var array
+	 */
+	protected $test_events = [];
+
+	/**
+	 * Data and meta data for the test posts we'll be using.
+	 *
+	 * @var array
+	 */
+	protected $post_example_settings;
+
+	/**
+	 * Set some default/common event meta to be used across test events.
+	 *
+	 * @since TBD
+	 */
+	public function setUp() {
+		// This needs to come first so that the post_example_settings template is created
+		parent::setUp();
+
+		$this->post_example_settings = array(
+			'post_author'           => 3,
+			'post_title'            => 'Unticketed Cost Formatting Test Event ',
+			'post_content'          => 'Testing out cost formatting on events when neither Event Tickets nor Event Tickets Plus are active.',
+			'post_status'           => 'publish',
+			'EventAllDay'           => false,
+			'EventHideFromUpcoming' => true,
+			'EventOrganizerID'      => 5,
+			'EventVenueID'          => 8,
+			'EventShowMapLink'      => true,
+			'EventShowMap'          => true,
+			'EventStartDate'        => '2012-01-01',
+			'EventEndDate'          => '2012-01-03',
+			'EventStartHour'        => '01',
+			'EventStartMinute'      => '15',
+			'EventStartMeridian'    => 'am',
+			'EventEndHour'          => '03',
+			'EventEndMinute'        => '25',
+			'EventEndMeridian'      => 'pm',
+			'EventCurrencySymbol'   => '$'
+		);
+
+		$costs = [
+			null, // A null value means the event has no cost, as distinct from being "Free".
+			0,    // Should output "Free" string.
+			'4,92',
+			'4,999',
+			'5',
+			'5.99',
+			'25',
+			'100',
+			'180.067',
+			'3.00 8.00 125.95', // Represents a range of values.
+			'Free',             // The cost field should not manipulate this, and just output it literally.
+			'*&^$@#%@',         // Ditto: Should not be manipulated, just output literally.
+			'東京は都会です',     // Ditto: Should not be manipulated, just output literally.
+			'1995.95',
+		];
+
+		$iterations = 0;
+
+		foreach ( $costs as $event_cost ) {
```

Comment:
Why not extend the `Events_TestCase` and use its very convenient factory? All this code would collapse into one line.
`$this->factory()->event->create(['meta_input' => ['_EventCost' => 5]]` and so on.

Reply:
Plus: not a fan of too smart setup methods when the setup can be shown in the test methods: ideally a test method contains Arrange, Act, Assert. Whenever you move one of the steps out of the test method it loses clarity.